core/performance_test/peakflops/02_kokkos_simple_view.cpp

```diff
@@ -0,0 +1,189 @@
+
+view_type //__attribute__ ((noinline))
+axpy_10(
+        const view_type & __restrict__  a,
+        const view_type & __restrict__  x0,
+        const view_type & __restrict__  x1,
+        const view_type & __restrict__  x2,
+        const view_type & __restrict__  x3,
+        const view_type & __restrict__  x4,
+        const view_type & __restrict__  x5,
+        const view_type & __restrict__  x6,
+        const view_type & __restrict__  x7,
+        const view_type & __restrict__  x8,
+        const view_type & __restrict__  x9,
+        const view_type & __restrict__  c,
+        long n
+)
+{
+    long i;
+    int j;
+
+    asm volatile ("# ax+c loop begin");
+    for(i = 0; i<n; i++)
+    {
+#pragma omp simd
+        for(j=0; j<VECLENTH; j++)
+        {
+            x0(j) = a(j)*x0(j)+ c(j);
+            x1(j) = a(j)*x1(j)+ c(j);
+            x2(j) = a(j)*x2(j)+ c(j);
+            x3(j) = a(j)*x3(j)+ c(j);
+            x4(j) = a(j)*x4(j)+ c(j);
+            x5(j) = a(j)*x5(j)+ c(j);
+            x6(j) = a(j)*x6(j)+ c(j);
+            x7(j) = a(j)*x7(j)+ c(j);
+            x8(j) = a(j)*x8(j)+ c(j);
+            x9(j) = a(j)*x9(j)+ c(j);
+        }
+    }
+    asm volatile ("# ax+c loop end");
+
+    for(j=0; j<VECLENTH; j++){
+        x0(j) = x0(j)+x1(j)+x2(j)+x3(j)+x4(j)+x5(j)+x6(j)+x7(j)+x8(j)+x9(j)+(float)n;
+    }
+    return x0;
+}
+
+
+TEST(kokkos, simple) {
+    long n = static_cast<long>(2e6); //1000000000; //MAXBYTES/sizeof(float);
+    /*long seed = (argc > 2 ? atol(argv[2]) : 76843802738543);*/
+    long seed = 76843802738543;
+
+    const int N = 1;
+
+    view_type a_("a",N);
+    view_type x0_("x0",N);
+    view_type x1_("x1",N);
+    view_type x2_("x2",N);
+    view_type x3_("x3",N);
+    view_type x4_("x4",N);
+    view_type x5_("x5",N);
+    view_type x6_("x6",N);
+    view_type x7_("x7",N);
+    view_type x8_("x8",N);
+    view_type x9_("x9",N);
+    view_type c_("c",N);
+
+    /*
+    data * a_=new data;
+    data * x_=new data;
+    data * x1_=new data;
+    data * x2_=new data;
+    data * x3_=new data;
+    data * x4_=new data;
+    data * x5_=new data;
+    data * x6_=new data;
+    data * x7_=new data;
+    data * x8_=new data;
+    data * x9_=new data;
+    data * c_=new data;
+    */
+
+    long i,j;
+    unsigned short rg[3] = { static_cast<unsigned short>(seed >> 16), static_cast<unsigned short>(seed >> 8), static_cast<unsigned short>(seed) };
+
+    for (i = 0; i < VECLENTH; i++)
+    {
+        a_(i)  = erand48(rg);
+        c_(i)  = erand48(rg);
+        x0_(i)  = erand48(rg);
+        x1_(i) = erand48(rg);
+        x2_(i) = erand48(rg);
+        x3_(i) = erand48(rg);
+        x4_(i) = erand48(rg);
+        x5_(i) = erand48(rg);
+        x6_(i) = erand48(rg);
+        x7_(i) = erand48(rg);
+        x8_(i) = erand48(rg);
+        x9_(i) = erand48(rg);
+    }
+    // for (i = 0; i < VECLENTH; i++) {
```

Comment:
Is this for debug purposes? If so you may want to place in some kind of debug print macro.

Reply:
fixed